include/dump_utils.hpp

```diff
@@ -0,0 +1,110 @@
+#pragma once
+
+#include "async_resp.hpp"
+#include "error_messages.hpp"
+#include "logging.hpp"
+#include "utility.hpp"
+
+#include <boost/algorithm/string/case_conv.hpp>
+#include <boost/system/error_code.hpp>
+#include <dbus_singleton.hpp>
+#include <dbus_utility.hpp>
+
+#include <cstddef>
+#include <string>
+
+namespace redfish
+{
+namespace dump_utils
+{
+
+inline void getValidDumpEntryForAttachment(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp, const std::string& url,
+    std::function<void(const std::string& objectPath,
+                       const std::string& entryID,
+                       const std::string& dumpType)>&& callback)
+{
+    std::string dumpType;
+    std::string entryID;
+    std::string entriesPath;
+
+    if (crow::utility::readUrlSegments(url, "redfish", "v1", "Managers", "bmc",
+                                       "LogServices", "Dump", "Entries",
+                                       std::ref(entryID), "attachment"))
+    {
+        // BMC type dump
+        dumpType = "BMC";
+        entriesPath = "/redfish/v1/Managers/bmc/LogServices/Dump/Entries/";
+    }
+    else if (crow::utility::readUrlSegments(
+                 url, "redfish", "v1", "Systems", "system", "LogServices",
+                 "Dump", "Entries", std::ref(entryID), "attachment"))
+    {
+        // System type
+        dumpType = "System";
+        entriesPath = "/redfish/v1/Managers/bmc/LogServices/Dump/Entries/";
```

Comment:
for system dump entries path should be redfish path should be
/redfish/v1/Systems/system/Logservices/Dump/Entries

Reply:
entriesPath for System dump is changed to
`entriesPath = "/redfish/v1/Systems/system/LogServices/Dump/Entries/";`

@raviteja-b 
BTW, how can I test system/resource dump?
Even after I initiating the system dump from GUI, I can't find any generated System dumps (after a while).

Please provide me the test instruction `for system/resource dump as well and post results`

Reply:
> for system dump entries path should be redfish path should be /redfish/v1/Systems/system/Logservices/Dump/Entries

Fixed

Reply:
@raviteja-b

This PR is for GET of the existing dumps.  

I think the team who owns DUMP would need to go thru POST part, or provide me the instruction how to do generate it.

Reply:
I've tested it using GUI system dump.

With the fix.
```
% curl -k -X GET https://${bmc}:18080/redfish/v1/Systems/system/LogServices/Dump/Entries/INVALID/attachment
{
  "error": {
    "@Message.ExtendedInfo": [
      {
        "@odata.type": "#Message.v1_1_1.Message",
        "Message": "The object at '/redfish/v1/Systems/system/LogServices/Dump/Entries/INVALID' is invalid.",
        "MessageArgs": [
          "/redfish/v1/Systems/system/LogServices/Dump/Entries/INVALID"
        ],
        "MessageId": "Base.1.13.0.InvalidObject",
        "MessageSeverity": "Critical",
        "Resolution": "Either the object is malformed or the URI is not correct.  Correct the condition and resubmit the request if it failed."
      }
    ],
    "code": "Base.1.13.0.InvalidObject",
    "message": "The object at '/redfish/v1/Systems/system/LogServices/Dump/Entries/INVALID' is invalid."
  }
}%
```